src/swupd/manifest.go

```diff
@@ -226,3 +226,113 @@ func (m *Manifest) ReadManifestFromFile(f string) error {
 	// return err so the deferred close can modify it
 	return err
 }
+
+// getManifestTimeStamp returns the unix timestamp of m.Header.TimeStamp
+func (m *Manifest) getManifestTimeStamp() int64 {
+	return m.Header.TimeStamp.Unix()
+}
+
+// writeManifestFileHeader writes the header of a manifest to the file
+func writeManifestFileHeader(m *Manifest, w *bufio.Writer) error {
```

Comment:
Should be able to just declare w as *io.Writer, no need to require it to be a *bufio.Writer.

Reply:
More precisely as: w io.Writer (without the pointer).

This function is making assumptions about how w handles errors (which hold true for *bufio.Writer but not for an arbitrary io.Writer), so you'll have to do more changes to handle errors for writes.

To make this less noisy, you can merge the prints into a single one:

    _, err := fmt.Fprintf(w, `MANIFEST\t%v
    version:\t%v
    ...`, m.Header.Format, ...)